src/components/Calculator.js

```diff
@@ -3,88 +3,80 @@ import React from 'react';
-        <div className="zero-button calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '0'); }} onKeyPress={() => { this.calculate(calculatorObj, '0'); }}>0</div>
-        <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '.'); }} onKeyPress={() => { this.calculate(calculatorObj, '.'); }}>.</div>
-        <div className="operators-buttons calc-button-style" role="button" tabIndex="0" onClick={() => { this.calculate(calculatorObj, '='); }} onKeyPress={() => { this.calculate(calculatorObj, '='); }}>=</div>
+  return (
+    <div id="calculator-container" className="calculator-container">
+      <div id="result" className="result-display calc-button-style">
+        { displayResult(calculatorObj) }
       </div>
-    );
-  }
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, 'AC', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, 'AC', setCalculatorObj); }}>AC</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '+/-', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '+/-', setCalculatorObj); }}>+/-</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '%', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '%', setCalculatorObj); }}>%</div>
+      <div className="operators-buttons calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '÷', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '÷', setCalculatorObj); }}>÷</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '7', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '7', setCalculatorObj); }}>7</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '8', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '8', setCalculatorObj); }}>8</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '9', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '9', setCalculatorObj); }}>9</div>
+      <div className="operators-buttons calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, 'x', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, 'x', setCalculatorObj); }}>x</div>
+      <div className="calc-button-style" role="button" tabIndex="0" onClick={() => { updateCalculation(calculatorObj, '4', setCalculatorObj); }} onKeyPress={() => { updateCalculation(calculatorObj, '4', setCalculatorObj); }}>4</div>
```

Comment:
**Optional Changes**
- [ ] There is a lot of repetition going on here. You could refactor this into it's own Button function Component. Also you could store the button names in an array and use `array.map` to dynamically add all the buttons to the DOM. This will allow us to stick to the `DRY` principle.

Reply:
Hello, @oxenprogrammer I'll do these changes I was lazy when I did this and realized just after a while that I did a bad job on this! Thank you for pointing out these changes as optional!